src/os/ui/layer/defaultlayerui.js

```diff
@@ -433,6 +433,16 @@ os.ui.layer.DefaultLayerUICtrl.prototype.onRefreshChange = function() {
  */
 os.ui.layer.DefaultLayerUICtrl.prototype.reset = function(key) {
   var defaultValue = this.defaults[key];
+  var nodeLabel = this.getLayerNodes()[0].label;
+  if (nodeLabel) {
+    nodeLabel = nodeLabel.replace(/ /g, '');
+    nodeLabel = nodeLabel.toLowerCase();
+    var value = /** @type {number} */ (os.settings.get(['providers', 'basemap', 'maps', nodeLabel, key]));
```

Comment:
Layers can come from many more providers than just the `BaseMapProvider`, whose settings you have hard coded here. A better way to do this would be to have the thing making the layer config (generally either the provider or descriptor) store the default value on the config object. The best place to do that is probably in `os.data.LayerSyncDescriptor.prototype.getOptions`, just before the user layer config is applied. Then this function can loop over each layer's `getLayerOptions()['defaultOpacity']` or whatever you end up calling it.

Note that there is no longer a single default value if multiple layers are selected. Each layer will (potentially) have its own default and that will need to be addressed as well.